src/caret_analyze/infra/lttng/records_provider_lttng.py

```diff
@@ -1241,17 +1313,26 @@ def to_records(self):
         assert self._node_path.subscription is not None and self._node_path.publisher is not None
 
         sub_records = self._provider.subscribe_records(self._node_path.subscription)
+        is_take_node = len(sub_records) == 0
+        if is_take_node:
+            sub_records = self._provider.subscription_take_records(self._node_path.subscription)
+            # TODO: Currently, pub_records and sub_records are being joined by source_timestamp.
+            # If possible, use rmw_take as the key.
         pub_records = self._provider.publish_records(self._node_path.publisher)
```

Comment:
pub_records and sub_records should be merged by merge_sequential() with rmw_take_timestamp specified in left_stamp_key.
This is a task that should be done now, not in a future revision.
It is expected that enough information for merging has already been obtained to prevent incorrect merging.

Reply:
delete 'TODO:' in https://github.com/tier4/caret_analyze/pull/516/commits/17aef41f62a7458496ec59aad109934b880aa1ca.